views-core/src/main/java/io/micronaut/views/reactive/DefaultReactiveViewsRendererLocator.java

```diff
@@ -0,0 +1,140 @@
+                                                        @NonNull String contentType,
+                                                        @Nullable Object body) throws ViewNotFoundException {
+        Class<?> bodyClass = body != null ? body.getClass() : null;
+        ViewsRendererKey key = new ViewsRendererKey(view, contentType, bodyClass);
+        return Optional.ofNullable(viewsRendererMap.computeIfAbsent(key, (viewsRendererKey -> {
+            List<ReactiveViewsRenderer> viewsRenderers = resolveViewsRenderer(bodyClass, contentType);
+            if (viewsRenderers.isEmpty()) {
+                return null;
+            }
+            Optional<ReactiveViewsRenderer> result = viewsRenderers.stream()
+                    .filter(viewsRenderer -> viewsRenderer.exists(view))
+                    .findFirst();
+            if (result.isPresent()) {
+                return result.get();
+            }
+            throw new ViewNotFoundException("View [" + view + "] does not exist");
+        })));
+    }
+
+    /**
+     *
+     * @param bodyClass Response Body Class
+     * @param contentType Response Content Type
+     * @return List of {@link ViewsRenderer} which includes those which do not specify an {@link @Produces} annotation or
+     * whose {link @Produces} annotation value matches the response content type. The list is sorted. The order is those {@link ViewsRenderer} which
+     * type argument matches the response body class first and then ordered by {@link OrderUtil#COMPARATOR}.
+     */
+    @NonNull
+    private List<ReactiveViewsRenderer> resolveViewsRenderer(Class<?> bodyClass, @NonNull String contentType) {
+        return reactiveViewsRenderersByBodyClass(bodyClass)
+                .stream()
+                .filter(vr -> producesContentType(contentType, vr))
+                .sorted((o1, o2) -> {
+                    BeanDefinition o1BeanDefinition = beanDefinitionForViewRenderer(o1);
+                    BeanDefinition o2BeanDefinition = beanDefinitionForViewRenderer(o2);
+                    if (o1BeanDefinition.getTypeArguments().size() != o2BeanDefinition.getTypeArguments().size()) {
+                        return Integer.compare(o1BeanDefinition.getTypeArguments().size(), o2BeanDefinition.getTypeArguments().size());
+                    }
+                    return OrderUtil.COMPARATOR.compare(o1, o2);
+                }).toList();
+    }
+
+    private Collection<ReactiveViewsRenderer> reactiveViewsRenderersByBodyClass(Class<?> bodyClass) {
+        if (bodyClass == null) {
+            return beanContext.getBeansOfType(ReactiveViewsRenderer.class);
+        }
+        Collection<ViewsRenderer> viewsRenderers = beanContext.getBeansOfType(ViewsRenderer.class, Qualifiers.byTypeArguments(bodyClass, Object.class));
+        return beanContext.getBeansOfType(ReactiveViewsRenderer.class, Qualifiers.byTypeArguments(bodyClass, Object.class, Object.class))
```

Comment:
think it would be better to remove the generics and add a `supports(Class<?> type)` method to the `ReactiveViewsRenderer` rather than 2 getBeansOfType calls which looks like a hack

Reply:
I agree it is hackish. But we need to call twice because many `ReactiveViewsRenderer` implementations are backed by a `ViewsRenderer`, the ones using `ReactiveViewsRendererAdapter`. 
For those implementations, we need to know if the delegated class supports the body class.  

Note, that the lookup cost for the appropriate view renderer for a particular body class is paid only once as it is cached in a map.

Reply:
ok but I am unconvinced wrapping the non-reactive in a reactive API by default is the way to go. The `Writable` part not being shifted to the blocking thread pool is a problem

Reply:
it would be better to leave the `ViewRenderer` stuff as is and add an optional interface `ReactiveViewRenderer` that can be implemented by the view renderers that support it and then adapt the code with different `if/else` blocks the adding reactive overhead to all view renderers and changing the behaviour of Writable

Reply:
@sdelamo can you look at splitting this so we are not adding reactive overhead unnecessary for non-reactive view renderers?